hyperparameters_diffusion.txt
DIFFREC ML-1M
batch_size=400, dims=[200,600], lr=0.0001, noise_max=0.01, noise_min=0.0005, noise_scale=0.005, reweight=False, steps=5, weight_decay=0.0

DIFFREC FOURSQUARE_TKY
batch_size=400, dims=[1000], lr=0.0001, noise_max=0.01, noise_min=0.001, noise_scale=0.005, reweight=False, steps=40, weight_decay=0.0

DIFFREC BOOKS
batch_size=400, dims=[1000], lr=0.0001, noise_max=0.01, noise_min=0.001, noise_scale=0.005, reweight=True, steps=5, weight_decay=0.0

DIFFREC LASTFM
batch_size=400, dims=[200,600], lr=0.0001, noise_max=0.005, noise_min=0.001, noise_scale=0.005, reweight=True, steps=10, weight_decay=0.0


L-DIFFREC ML-1M
batch_size=400, in_dims=[300], lamda=0.01, lr1=0.001, lr2=0.001, mlp_dims=[300], steps=5, n_cate=2, noise_max=0.01, noise_min=0.0005, noise_scale=0.1

L-DIFFREC FOURSQUARE_TKY
batch_size=400, in_dims=[300], lamda=0.03, lr1=0.001, lr2=0.001, mlp_dims=[300], steps=2, n_cate=2, noise_max=0.005, noise_min=0.001, noise_scale=0.1

L-DIFFREC BOOKS
batch_size=400, in_dims=[300], lamda=0.03, lr1=0.001, lr2=0.001, mlp_dims=[300], steps=10, n_cate=2, noise_max=0.01, noise_min=0.005, noise_scale=0.005

L-DIFFREC LASTFM
batch_size=400, in_dims=[300], lamda=0.03, lr1=0.01, lr2=0.01, mlp_dims=[300], steps=10, n_cate=2, noise_max=0.01, noise_min=0.005, noise_scale=0.1


CDIFF4REC ML-1M
alpha=0.5, lr=0.0001, r_agg=avg, steps=10, topk=20, weight_decay=0.0

CDIFF4REC FOURSQUARE_TKY
alpha=0.9, lr=0.0001, r_agg=sim, steps=10, topk=20, weight_decay=0.0

CDIFF4REC BOOKS
alpha=0.9, lr=0.0001, r_agg=sim, steps=10, topk=20, weight_decay=0.0

CDIFF4REC LASTFM
alpha=0.7, lr=0.0001, r_agg=sim, steps=10, topk=20, weight_decay=0.0


GIFFCF ML-1M
ideal_weight=0, learning_rates=[0.001, 0.001], noise_decay=0.5

GIFFCF FOURSQUARE_TKY
ideal_weight=0.3, learning_rates=[0.001, 0.001], noise_decay=0.5

GIFFCF BOOKS
ideal_weight=0, learning_rates=[0.001, 0.001], noise_decay=0.5

GIFFCF LASTFM
ideal_weight=0.3, learning_rates=[0.001, 0.001], noise_decay=1


CONDIFF ML-1M
dims=[200,200], lamd1=0.1, lamd2=0.1, lr=0.0005, noise_max=0.2, noise_min=0.02, noise_scale=0.001, steps=20, weight_decay=0.01

CONDIFF FOURSQUARE_TKY
dims=[1000], lamd1=0.1, lamd2=0.1, lr=5e-05, noise_max=0.01, noise_min=0.001, noise_scale=0.001, steps=20, weight_decay=0.01

CONDIFF BOOKS
dims=[1000], lamd1=0.1, lamd2=0.1, lr=5e-05, noise_max=0.2, noise_min=0.001, noise_scale=0.001, steps=20, weight_decay=0.01

CONDIFF LASTFM
dims=[200,200], lamd1=0.1, lamd2=0.1, lr=0.0005, noise_max=0.01, noise_min=0.001, noise_scale=0.001, steps=10, weight_decay=0.01


CODIGEM ML-1M
M=1000, T=3, beta=0.0001, lr=1e-3

CODIGEM FOURSQUARE_TKY
M=1000, T=5, beta=0.0001, lr=1e-3

CODIGEM BOOKS
M=1000, T=3, beta=0.0001, lr=1e-3

CODIGEM LASTFM
M=200, T=3, beta=0.0001, lr=1e-3


CF-DIFF ML-1M
lr=0.0001, noise_max=0.01, noise_min=0.001, noise_scale=0.001, steps=20

CF-DIFF FOURSQUARE_TKY
lr=0.0001, noise_max=0.01, noise_min=0.001, noise_scale=0.001, steps=20

CF-DIFF BOOKS
lr=0.001, noise_max=0.01, noise_min=0.0001, noise_scale=0.0001, steps=10

CF-DIFF LASTFM
lr=0.001, noise_max=0.01, noise_min=0.0001, noise_scale=0.00001, steps=10


BSPM-LM ML-1M
T_s=2.25, final_sharpening=False, idl_beta=0.15, simple_model=bspm, solver_shr=rk4

BSPM-EM ML-1M
T_s=2.25, final_sharpening=True, idl_beta=0.15, simple_model=bspm, solver_shr=rk4

BSPM-LM FOURSQUARE_TKY
T_s=1, final_sharpening=False, idl_beta=0.15, simple_model=bspm, solver_shr=rk4

BSPM-EM FOURSQUARE_TKY
T_s=2.25, final_sharpening=True, idl_beta=0.15, simple_model=bspm, solver_shr=rk4

BSPM-LM BOOKS
T_s=2.25, final_sharpening=False, idl_beta=0.15, simple_model=bspm, solver_shr=rk4

BSPM-EM BOOKS
T_s=2.5, final_sharpening=True, idl_beta=0.15, simple_model=bspm, solver_shr=rk4

BSPM-LM LASTFM
T_s=2, final_sharpening=False, idl_beta=0.0, simple_model=bspm, solver_shr=rk4

BSPM-EM LASTFM
T_s=2, final_sharpening=True, idl_beta=0.0, simple_model=bspm, solver_shr=rk4

hyperparameters_traditional.txt
BPRMF ML-1M
embedding_size: 128, learning_rate: 0.0002, train_batch_size: 2048

BPRMF FTKY
embedding_size: 128, learning_rate: 0.0002, train_batch_size: 2048

BPRMF BOOKS
embedding_size: 128, learning_rate: 0.0001, train_batch_size: 1024

BPRMF LASTFM
embedding_size: 128, learning_rate: 0.0001, train_batch_size: 2048


ItemKNN ML-1M
k: 400, shrink: 2

ItemKNN FTKY
k: 200, shrink: 0

ItemKNN BOOKS
k: 100, shrink: 2.0

ItemKNN LASTFM
k: 200, shrink: 0.0


NeuMF ML-1M
dropout_prob: 0.3, learning_rate: 0.0002, mf_embedding_size: 64, mlp_hidden_size: [128,64], train_batch_size: 2048

NeuMF FTKY
dropout_prob: 0.3, learning_rate: 0.0002, mf_embedding_size: 128, mlp_hidden_size: [128,64], train_batch_size: 2048

NeuMF BOOKS
dropout_prob: 0.1, learning_rate: 0.0001, mf_embedding_size: 128, mlp_hidden_size: [128,64], train_batch_size: 1024

NeuMF LASTFM
dropout_prob: 0.3, learning_rate: 0.0001, mf_embedding_size: 128, mlp_hidden_size: [128,64], train_batch_size: 2048


LightGCN ML-1M
learning_rate: 0.0002, embedding_size: 64, n_layers: 2, train_batch_size: 2048

LightGCN FTKY
learning_rate: 0.0002, embedding_size: 64, n_layers: 2, train_batch_size: 2048

LightGCN BOOKS
learning_rate: 0.001, embedding_size: 128, n_layers: 2, train_batch_size: 2048

LightGCN LASTFM
learning_rate: 0.001, embedding_size: 128, n_layers: 3, train_batch_size: 2048


UltraGCN ML-1M
ILoss_lambda: 1e-7, learning_rate: 2e-4, negative_weight: 10, w1: 1.0, w2: 1e-7, w3: 1.0, w4: 1e-7

UltraGCN FTKY
ILoss_lambda: 1e-7, learning_rate: 2e-4, negative_weight: 10, w1: 1.0, w2: 1e-7, w3: 1.0, w4: 1e-7

UltraGCN BOOKS
ILoss_lambda: 1e-4, learning_rate: 5e-4, negative_weight: 5, w1: 1.0, w2: 1.0, w3: 1.0, w4: 1.0

UltraGCN LASTFM
ILoss_lambda: 1e-4, learning_rate: 5e-4, negative_weight: 10, w1: 1.0, w2: 1.0, w3: 1.0, w4: 1.0


XSimGCL ML-1M
eps: 0.2, lambda: 0.05, learning_rate: 2e-4, temperature: 0.1

XSimGCL FTKY
eps: 0.2, lambda: 0.05, learning_rate: 2e-4, temperature: 0.1

XSimGCL BOOKS
eps: 0.2, lambda: 0.05, learning_rate: 5e-4, temperature: 0.1

XSimGCL LASTFM
eps: 0.2, lambda: 0.05, learning_rate: 1e-4, temperature: 0.1


EASE ML-1M
reg_weight: 2000

EASE FTKY
reg_weight: 100

EASE BOOKS
reg_weight: 100

EASE LASTFM
reg_weight: 1000


MultiVAE ML-1M
dropout_prob: 0.3, mlp_hidden_size: [600], learning_rate: 2e-4, latent_dimension: 200

MultiVAE FTKY
dropout_prob: 0.5, mlp_hidden_size: [600], learning_rate: 2e-4, latent_dimension: 200

MultiVAE BOOKS
dropout_prob: 0.5, mlp_hidden_size: [600], learning_rate: 1e-3, latent_dimension: 200

MultiVAE LASTFM
dropout_prob: 0.3, mlp_hidden_size: [600], learning_rate: 1e-3, latent_dimension: 200


RecVAE ML-1M
dropout_prob: 0.1, hidden_dimension: 600, gamma: 5e-3, learning_rate: 2e-4

RecVAE FTKY
dropout_prob: 0.5, hidden_dimension: 600, gamma: 5e-3, learning_rate: 2e-4

RecVAE BOOKS
dropout_prob: 0.5, hidden_dimension: 600, gamma: 5e-3, learning_rate: 1e-3

RecVAE LASTFM
dropout_prob: 0.5, hidden_dimension: 600, gamma: 5e-3, learning_rate: 5e-4

CDiff4Rec/config.py
import argparse

def get_config():
    parser = argparse.ArgumentParser()
    parser.add_argument('--dataset', type=str, default='yelp_created', help='choose the dataset')
    parser.add_argument('--data_path', type=str, default='../datasets/', help='load data path')
    parser.add_argument('--lr', type=float, default=0.0001, help='learning rate')
    parser.add_argument('--weight_decay', type=float, default=0.0)
    parser.add_argument('--batch_size', type=int, default=400)
    parser.add_argument('--epochs', type=int, default=1000, help='upper epoch limit')
    parser.add_argument('--topN', type=str, default='[10, 20, 50, 100]')
    parser.add_argument('--tst_w_val', action='store_true', help='test with validation')
    parser.add_argument('--cuda', action='store_true', help='use CUDA')
    parser.add_argument('--gpu', type=str, default='0', help='gpu card ID')
    parser.add_argument('--save', "--s", action='store_true', help='test with validation')
    parser.add_argument('--save_path', "--sp", type=str, default='./saved_models/', help='save model path')
    parser.add_argument('--log_name', type=str, default='log', help='the log name')
    parser.add_argument('--round', type=int, default=1, help='record the experiment')

    # params for the model
    parser.add_argument('--time_type', type=str, default='cat', help='cat or add')
    parser.add_argument('--dims', type=str, default='[1000]', help='the dims for the DNN')
    parser.add_argument('--norm', type=bool, default=False, help='Normalize the input or not')
    parser.add_argument('--emb_size', type=int, default=10, help='timestep embedding size')

    # params for diffusion
    parser.add_argument('--mean_type', type=str, default='x0', help='MeanType for diffusion: x0, eps')
    parser.add_argument('--steps', type=int, default=50, help='diffusion steps')
    parser.add_argument('--noise_schedule', type=str, default='linear-var', help='the schedule for noise generating')
    parser.add_argument('--noise_scale', type=float, default=0.1, help='noise scale for noise generating')
    parser.add_argument('--noise_min', type=float, default=0.0001, help='noise lower bound for noise generating')
    parser.add_argument('--noise_max', type=float, default=0.02, help='noise upper bound for noise generating')
    parser.add_argument('--sampling_noise', type=bool, default=False, help='sampling with noise or not')
    parser.add_argument('--sampling_steps', type=int, default=0, help='steps of the forward process during inference')
    parser.add_argument('--reweight', type=bool, default=True, help='assign different weight to different timestep or not')

    # params for global attention
    parser.add_argument('--alpha', type=float, default=0.1, help='the importance of the user’s own preference')
    parser.add_argument('--beta', type=float, default=0.9, help='the significance of real user')
    parser.add_argument('--gamma', type=float, default=0, help='the significance of psuedo user')

    parser.add_argument('--tau', type=float, default=1.0, help='the temperature for consie similarity')
    parser.add_argument('--topk', type=int, default=10, help='he number of top-K similar interest users')
    parser.add_argument('--topk_pseudo_user', type=int, default = 0, help='quantity of pseudo-users')

    parser.add_argument('--using_real', "--ur", action = argparse.BooleanOptionalAction, default=True, help = "whether using real users (--ur or --no-ur)")
    parser.add_argument('--using_pseudo', "--uf", action = argparse.BooleanOptionalAction, default=False, help = "whether using psuedo-users (--uf or --no-uf)")
    parser.add_argument('--training_pseudo', "--tf", action = argparse.BooleanOptionalAction, default=False, help = "whether training psuedo-users or not (--tf or --no-tf)")
    parser.add_argument('--r_agg',type = str, default = None, help = "att or avg or sim")
    parser.add_argument('--f_agg',type = str, default = None, help = "att or avg or sim")

    parser.add_argument('--random_seed', "--rs", type=int, default=1, help='random_seed')
    parser.add_argument('--lamda', type=float, default = 0.001, help='weight for pseudo user loss')

    args = parser.parse_args()
    return args
